src/Microsoft.Health.Fhir.Core/Features/Conformance/SystemConformanceProvider.cs

```diff
@@ -351,27 +340,16 @@ public override async Task<ResourceElement> GetMetadata(CancellationToken cancel
             }
 
             // There is a chance that the BackgroundLoop handler sets _metadata to null between when it is checked and returned, so the value is stored in a local variable.
-            ResourceElement metadata;
-            if ((metadata = _metadata) != null)
+            ResourceElement metadata = await UpdateBuilderAsync(cancellationToken);
+            if (metadata == null)
```

Comment:
Is it possible to have a race condition here?

My thinking is multiple concurrent calls to GetMetadata() will all call UpdateBuilderAsync() without any locking, potentially triggering multiple expensive queries